src/Domain/UseCase/CreateValetToken/CreateValetTokenValidation.ts

```diff
@@ -0,0 +1,164 @@
+import { CreateValetKeyDto } from './CreateValetTokenDto'
+import { ValidatedValue } from '../../Validation/Validation'
+import { Request } from 'express'
+import { Operation } from '../../Operation/Operation'
+import { Resource } from '../../Resource/Resource'
+import { validateUuid } from '../../Uuid/validateUuid'
+import { UserPermission, UserPermissions, UserWithPermissions } from '../../User/UserWithPermissions'
+import { ValidityPeriod } from '../../ValetToken/ValetToken'
+import { dateFormat, maxExpiresAfterSeconds, minExpiresAfterSeconds } from '../../ValetToken/constants'
+import { validateDateString } from '../../Date/validateDateString'
+
+/**
+ * Assuming `request` contains a JS object in the `body`, parsed as JSON.
+ */
+export function validateCreateValetTokenRequest(request: Request): 
```

Comment:
There are two concerns mixed in here - validating data types and validating business logic. Can you leave just the validation of business logic in here and remove validation of data types like uuid etc? Data types can be validated on a different layer (HTTP) with for example `Ajv`

Reply:
I don't think there is anything but dynamic data type validation here -- I specifically was writing this with only this concern in mind. So there should be no business logic. If that's not the case then it's a mistake and I'll remove it if you point me to a specific place where this happens.

That said I'll consider this to be part of the same validation thread for which I created a task here: https://app.asana.com/0/1199914526147392/1200093443903073/f

Reply:
Resolving on the account of the discussion, task, and latest refactor.